types/firestore.d.ts

```diff
@@ -2063,18 +2094,62 @@ declare namespace FirebaseFirestore {
     toQuery(): Query<T>;
   }
 
+  /**
+   * Union type representing the aggregate type to be performed.
+   */
+  export type AggregateType = 'count' | 'avg' | 'sum';
+
+  // on the return type of `sum(..)`, `average(...)`, and `count()`
+  /**
+   * The union of all `AggregateField` types that are supported by Firestore.
+   */
+  export type AggregateFieldType = AggregateField<number | null>;
```

Comment:
This could be changed to improve the readability of the code. The original code below was specified in the API proposal. I'm open to your thoughts on this. It's also something that can be changed later.

```
type AggregateFieldType =
  ReturnType<typeof count>
  | ReturnType<typeof sum>
  | ReturnType<typeof average>;
```

Reply:
I intended to make this update (as the comment on 2102 suggests), but missed it. Done.